### PR TITLE
app

### PR DESCRIPTION
**Pull request by knehrybecki**

### CODE REVIEW

File: src/types/types.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export enum Difficulty {
+    Easy = 'Easy',
+    Medium = 'Medium',
+    Hard = 'Hard'
+}
+
+export type Question = {
+    response_code: number,
+    results: [{
```
Comment (jpudysz) [inline src/types/types.ts]: this should be a separate type, no-online-types

File: src/selectedCategory/selectedCategory.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import $ from 'jquery'
+import { quiz } from '..'
```
Comment (jpudysz) [inline src/selectedCategory/selectedCategory.ts]: bad import

File: src/selectedCategory/selectedCategory.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            name: 'easy',
+            categoryId: target.id,
+            text: Difficulty.Easy,
+        })
+            .appendTo(quizDifficult)
+            .hide()
+
+        const difficultMedium = $('<div>', {
+            class: 'difficulty',
+            name: 'medium',
```
Comment (jpudysz) [inline src/selectedCategory/selectedCategory.ts]: use enums

File: src/selectedCategory/selectedCategory.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            categoryId: target.id,
+            text: Difficulty.Medium,
+        })
+            .appendTo(quizDifficult)
+            .hide()
+
+        const difficultHard = $('<div>', {
+            class: 'difficulty',
+            name: 'hard',
```
Comment (jpudysz) [inline src/selectedCategory/selectedCategory.ts]: user enums

File: src/selectedCategory/selectedCategory.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export const selectedCategory = () => {
+    $('.category__list').click(event => {
+        $('.quiz--alert').remove()
+
+        const target = event.target
+        const quizDifficult = $('<div>', { class: 'quiz__difficulty' }).appendTo(quiz)
+
+        const difficultEasy = $('<div>', {
+            class: 'difficulty',
+            name: 'easy',
```
Comment (jpudysz) [inline src/selectedCategory/selectedCategory.ts]: use enums

File: src/selectedCategory/selectedDifficulty.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
+import he from 'he'
+import $, { error } from 'jquery'
+import Swal from 'sweetalert2'
+import { quiz } from '../'
```
Comment (jpudysz) [inline src/selectedCategory/selectedDifficulty.ts]: bad import

File: src/selectedCategory/selectedDifficulty.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        quizDifficulty.slideUp(1000)
+        quizText.slideUp(200)
+
+        setTimeout(() => $('.quiz--spinner').show(), 1000)
+
+        quizDifficulty.remove()
+        quizText.remove()
+        $('.quiz').addClass('questions')
+
+        const APIQUESTIONS = `https://opentdb.com/api.php?amount=10&category=${categoryID}&difficulty=${difficultName}&type=multiple`
```
Comment (jpudysz) [inline src/selectedCategory/selectedDifficulty.ts]: these are two words, so it should be `API_QUESTIONS`

File: src/selectedCategory/selectedDifficulty.ts [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+                    if (res.ok) {
+                        return res.json()
+                    }
+
+                    throw error
+                })
+                .then((question: Question) => {
+                    const { response_code, results } = question
+
+                    if (response_code === 0) {
```
Comment (jpudysz) [inline src/selectedCategory/selectedDifficulty.ts:38]: there is no response code = 0

File: src/selectedCategory/selectedDifficulty.ts [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+        setTimeout(() => $('.quiz--spinner').show(), 1000)
+
+        quizDifficulty.remove()
+        quizText.remove()
+        $('.quiz').addClass('questions')
+
+        const APIQUESTIONS = `https://opentdb.com/api.php?amount=10&category=${categoryID}&difficulty=${difficultName}&type=multiple`
+
+        setTimeout(() => {
```
Comment (jpudysz) [inline src/selectedCategory/selectedDifficulty.ts]: setTimeout + timeout? Something is wrong here ;)

File: src/checkAnswer/checkAnswer.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
+import 'animate.css'
+import $ from 'jquery'
+import Swal from 'sweetalert2'
+import { renderCategory } from '../renderQuiz'
+import { Question } from '../types'
+
+export const checkAnswer = (results: Question['results']) => {
```
Comment (jpudysz) [inline src/checkAnswer/checkAnswer.ts]: strange default parameter

File: src/checkAnswer/checkAnswer.ts [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    let questionNumber: number = 0
+    let Scores: number = 0
```
Comment (jpudysz) [inline src/checkAnswer/checkAnswer.ts]: variables should be lowercased

File: src/renderQuiz/renderQuiz.ts [hunk 12]
```diff
@@ -0,0 +1,87 @@
+import $, { error } from 'jquery'
+import Swal from 'sweetalert2'
+import 'sweetalert2/src/sweetalert2.scss'
+import { quiz } from '../'
```
Comment (jpudysz) [inline src/renderQuiz/renderQuiz.ts]: bad import

File: src/renderQuiz/renderQuiz.ts [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+    const spinner = $('<img>', {
+        class: 'quiz--spinner',
+        src: '/src/img/Spinner-1s-200px.gif',
+    }).appendTo(quiz)
+
+    const quizCategory = $('<div>', { class: 'quiz__category' }).appendTo(quiz)
+    const category = $('<div>', { class: 'category' }).appendTo(quizCategory)
+
+     fetch(APICATEGORY)
```
Comment (jpudysz) [inline src/renderQuiz/renderQuiz.ts]: API_CATEGORY

**Review by jpudysz — CHANGES_REQUESTED** [review]

File: src/selectedCategory/selectedDifficulty.ts (hunk 8 above)
Comment (knehrybecki) [reply src/selectedCategory/selectedDifficulty.ts:38]: this is api code to help the programmer

Response Codes
The API appends a "Response Code" to each API Call to help tell developers what the API is doing.

Code 0: Success Returned results successfully.
Code 1: No Results Could not return results. The API doesn't have enough questions for your query. (Ex. Asking for 50 Questions in a Category that only has 20.)
Code 2: Invalid Parameter Contains an invalid parameter. Arguements passed in aren't valid. (Ex. Amount = Five)
Code 3: Token Not Found Session Token does not exist.
Code 4: Token Empty Session Token has returned all possible questions for the specified query. Resetting the Token is necessary.

File: src/types/types.ts [hunk 14]
```diff
@@ -4,19 +4,21 @@ export type Categories = {
...
-        correct_answer: string,
-        difficulty: string,
-        incorrect_answers: Array<string>
-        question: string,
-        type: string
-    }]
+    results: Array<Result>
+}
+
+type Result = {
```
Comment (jpudysz) [inline src/types/types.ts]: `QuizResponse`

File: src/selectedCategory/selectedDifficulty.ts [hunk 15]
```diff
@@ -35,6 +33,7 @@ export const selectedDifficulty = () => {
                     throw error
                 })
                 .then((question: Question) => {
+                    console.log(question)
```
Comment (jpudysz) [inline src/selectedCategory/selectedDifficulty.ts]: remove console.log

**Review by jpudysz — CHANGES_REQUESTED** [review]

**Pull request merged**